packages/dmn-js-decision-table/src/Viewer.js

```diff
@@ -36,31 +36,62 @@ export default class Viewer extends Table {
     this._container = container;
   }
 
-  open(decision, done) {
-
-    var err;
-
-    // use try/catch to not swallow synchronous exceptions
-    // that may be raised during model parsing
-    try {
+  /**
+   * @typedef {Object} OpenResult
+   * @property {Array<string>} warnings - Warnings occured during opening
+   */
 
-      if (this._decision) {
+  /**
+    * @typedef {Object} OpenError
+    * @property {Error} error
+    * @property {Array<string>} warnings - Warnings occured during opening
+    */
 
-        // clear existing rendered diagram
-        this.clear();
+  /**
+   * Open diagram element.
+   *
+   * @param  {ModdleElement} decision
+   * @returns {Promise} Resolves with {OpenResult} when successful
+   * or rejects with {OpenError}
+   */
+  open(decision) {
```

Comment:
So we won't have this fallback wrapper for this (at least for a while), right? Will we have it for other APIs of dmn-js or do we completely follow the update route here?

https://github.com/bpmn-io/bpmn-js/blob/c3e0d6d16477bc2e9e37109a9ff1254b6695c7cc/lib/util/CompatibilityUtil.js#L13

Reply:
Yes we will have it, but only for the key public APIs:

![Screenshot_20210608_151935](https://user-images.githubusercontent.com/42800119/121192160-f0636100-c86c-11eb-8be0-a4ededef1d4f.png)

Reply:
In bpmn-js, we also do it for other APIs, as [`open`](https://github.com/bpmn-io/bpmn-js/blob/develop/lib/BaseViewer.js#L279). The question is whether we want to have it the same for dmn-js.

Reply:
thanks for highlighting. I will look into this.

Reply:
I added the `wrapForCompatability` around `Manager#open`.

Reason:
* `Manager#open` is the public entry point when using dmn-js and wanting to open a View
  * We expect library users to use `dmn-js` as a whole. (e.g., loading the diagram using dmn-js#importXML, then switching view with `dmn-js#open` which goes via the `Manager`)
  * Although using specific views individually, this is (IMO) not a supported public API use-case (e.g., see that we don't tests these APIs stand-alone, also there are no examples around this)
  * Therefore I propose not to `wrapForCompatability` in the specific Viewers, since this would only blow-up the solution.

Happy to discuss

Reply:
Makes absolute sense. :+1: